src/hera/workflows/script.py

```diff
@@ -407,9 +407,11 @@ def _get_param_script_portion(self, instance: Script) -> str:
             # non-JSON encoded strings are returned, which fail the loads, but they can be used as plain strings
             # which is why this captures that in an except. This is only used for `InputFrom` cases as the extra
             # payload of the script is not necessary when regular input is set on the task via `func_params`
-            extract += f"""try: {param.name} = json.loads(r'''{{{{inputs.parameters.{param.name}}}}}''')\n"""
-            extract += f"""except: {param.name} = r'''{{{{inputs.parameters.{param.name}}}}}'''\n"""
-        return textwrap.dedent(extract)
+
+            if param.value_from is None:
+                extract += f"""try: {param.name} = json.loads(r'''{{{{inputs.parameters.{param.name}}}}}''')\n"""
+                extract += f"""except: {param.name} = r'''{{{{inputs.parameters.{param.name}}}}}'''\n"""
+        return textwrap.dedent(extract) if extract != "import json\n" else ""
```

Comment:
I decided to not append the `import json\n` when we don't actually have any params. @elliotgunton suggested some users might use the implicit import though. Is this a problem?

Reply:
I think it's ok to skip `json` here. Hera assumes it is imported for Hera specifically, not for the user. If users do need `json` they are likely to explicitly import it without awareness of the implicit one